extension/page-detect.js

```diff
@@ -0,0 +1,53 @@
+'use strict';
+
+window.pageDetect = (() => {
+	const isGistCheck = location.hostname === 'gist.github.com';
+
+	const isGist = () => isGistCheck;
+
+	const isDashboard = () => location.pathname === '/' || /^(\/orgs\/[^\/]+)?\/dashboard/.test(location.pathname);
+
+	const isRepo = () => !isGist() && /^\/[^/]+\/[^/]+/.test(location.pathname);
```

Comment:
Do you think we should modify the `isRepo()` check so we guards us against things like `github.com/orgs/organisationname`, `github.com/settings/profile` and the like?